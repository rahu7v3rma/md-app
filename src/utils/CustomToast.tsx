import React from 'react';
import { StyleSheet, TouchableOpacity, View } from 'react-native';
import Toast, { ToastConfigParams } from 'react-native-toast-message';

import { CheckIcon, CrossIcon } from '@/assets/svgs';
import { Text } from '@/shared';
import { Size } from '@/shared/text';
import { Colors } from '@/theme/colors';

const toastConfig = {
    successResponse: ({ text1 }: ToastConfigParams<any>) => (
        <View style={[styles.container, styles.successContainer]}>
            <View style={styles.errorIconStyle}>
                <CheckIcon width={14} height={14} color={Colors.text.black} />
            </View>
            <Text
                style={styles.titleStyle}
                size={Size.XXSmall}
                color={Colors.text.white}
                fontWeight="600"
            >
                {text1}
            </Text>
        </View>
    ),
    errorResponse: ({ text1 }: ToastConfigParams<any>) => (
        <View style={[styles.container, styles.errorContainer]}>
            <TouchableOpacity
                style={styles.errorIconStyle}
                onPress={() => Toast.hide()}
            >
                <CrossIcon width={12} height={12} color={Colors.text.error} />
            </TouchableOpacity>
            <Text
                style={styles.titleStyle}
                size={Size.XXSmall}
                color={Colors.text.white}
            >
                {text1}
            </Text>
        </View>
    ),
    info: ({ text1 }: ToastConfigParams<any>) => (
        <View style={[styles.container, styles.infoContainer]}>
            <Text
                style={styles.titleStyle}
                size={Size.XXSmall}
                color={Colors.text.white}
            >
                {text1}
            </Text>
        </View>
    )
};

export default toastConfig;

const styles = StyleSheet.create({
    titleStyle: {
        marginLeft: 15
    },
    successIconStyle: {
        width: 25,
        height: 25,
        borderRadius: 25,
        backgroundColor: Colors.extras.white,
        justifyContent: 'center',
        alignItems: 'center'
    },
    errorIconStyle: {
        width: 17,
        height: 17,
        borderRadius: 17,
        backgroundColor: Colors.extras.white,
        justifyContent: 'center',
        alignItems: 'center'
    },
    container: {
        height: 55,
        width: '90%',
        borderRadius: 20,
        flexDirection: 'row',
        alignItems: 'center',
        paddingHorizontal: 16,
        marginBottom: 65
    },
    successContainer: {
        backgroundColor: Colors.toast.successBackground
    },
    errorContainer: {
        backgroundColor: Colors.toast.errorBackground
    },
    infoContainer: {
        backgroundColor: Colors.toast.infoBackground
    }
});
