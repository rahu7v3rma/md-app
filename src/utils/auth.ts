import { CommonActions } from '@react-navigation/native';
import * as Keychain from 'react-native-keychain';

import { navigationRef } from '@/navigation';
import { resetNotifications } from '@/services/notification';
import { resetStore } from '@/store';

export const setAuthToken = (token: string) => {
    Keychain.setGenericPassword('token', token);
};

export const getAuthToken = (): Promise<string | null> => {
    return Keychain.getGenericPassword().then((credentials) =>
        credentials === false ? null : credentials.password
    );
};

export const AUTHORIZATION_HEADER_NAME = 'X-Authorization';

export const getAuthorizationHeaderValue = () => {
    return getAuthToken()
        .then((authToken) => `Token ${authToken || ''}`)
        .catch(() => undefined);
};

export const resetAuthToken = () => {
    Keychain.resetGenericPassword();
};

export const logoutAction = (skipNavigation: boolean = false) => {
    resetStore();
    resetNotifications();
    resetAuthToken();

    if (!skipNavigation) {
        navigationRef.dispatch(
            CommonActions.reset({
                index: 0,
                routes: [{ name: 'SignIn' }]
            })
        );
    }
};
