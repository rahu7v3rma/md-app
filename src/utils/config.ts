import config from 'react-native-config';

export const API_BASE_URL = config.API_BASE_URL;
