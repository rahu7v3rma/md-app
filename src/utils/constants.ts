import { ReactionData } from 'stream-chat-react-native';

import logActivity from '@/assets/svg/logActivity.svg';
import logBlood from '@/assets/svg/logBlood.svg';
import logFast from '@/assets/svg/logFastIcon.svg';
import logFood from '@/assets/svg/logFood.svg';
import logHydration from '@/assets/svg/logHydration.svg';
import logInsulin from '@/assets/svg/logInsulin.svg';
import logMedication from '@/assets/svg/logMedication.svg';
import logWeight from '@/assets/svg/logWeight.svg';
import {
    Heart,
    HighFive,
    Like,
    Muscle,
    SadFace,
    SmileyFace
} from '@/assets/svgs';

export const DateTimeFormat = 'DD/MM/YYYY HH:mm';

export const Constants: any = {
    trackerTitle: {
        tracked: 'Fast tracking is Paused',
        complete: 'Fast was completed',
        tracking: 'Fast in progress'
    },
    notificationType: {
        STREAM_CHAT_MESSAGE: 'message.new',
        LOG_REMINDER: 'log.reminder',
        LESSON_REMINDER: 'lesson.reminder',
        FAST_COMPLETE: 'fast.complete'
    },
    logs: {
        userFast: 'UserFast',
        userWeight: 'UserWeight',
        userInsulin: 'UserInsulin',
        userExercise: 'UserExercise',
        userGlucose: 'UserGlucose',
        userDrink: 'UserDrink',
        userMedication: 'UserMedication',
        userFood: 'UserFood',
        userLesson: 'UserLesson'
    },
    logTypeDetails: {
        UserGlucose: {
            icon: logBlood,
            title: 'Glucose',
            screen: 'LogBlood'
        },
        UserMedication: {
            icon: logMedication,
            title: 'Medication',
            screen: 'LogMedication'
        },
        UserDrink: {
            icon: logHydration,
            title: 'Drink',
            screen: 'LogWaterIntake'
        },
        UserWeight: {
            icon: logWeight,
            screen: 'LogWeight',
            title: 'Weight'
        },
        UserFast: {
            icon: logFast,
            title: 'Fast',
            screen: 'LogFast'
        },
        UserExercise: {
            icon: logActivity,
            title: 'Activity',
            screen: 'LogActivity'
        },
        UserInsulin: {
            icon: logInsulin,
            title: 'Insulin',
            screen: 'LogInsulin'
        },
        UserFood: {
            icon: logFood,
            title: 'Food',
            screen: 'LogMeal'
        },
        UserLesson: {
            icon: logActivity,
            title: 'Lesson Completed',
            screen: 'LessonContent'
        }
    },
    logTabsList: [
        {
            id: 1,
            icon: logBlood,
            title: 'Log Blood Glucose',
            screen: 'LogBlood',
            type: 'UserGlucose'
        },
        {
            id: 2,
            icon: logHydration,
            title: 'Log Hydration',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 3,
            icon: logActivity,
            title: 'Log Activity',
            screen: 'LogActivity',
            type: 'UserExercise'
        },
        {
            id: 4,
            icon: logFood,
            title: 'Log a Meal',
            screen: 'LogMeal',
            type: 'UserFood'
        },
        {
            id: 5,
            icon: logMedication,
            title: 'Log Medication',
            screen: 'LogMedication',
            type: 'UserMedication'
        },
        {
            id: 6,
            icon: logWeight,
            title: 'Log Weight',
            screen: 'LogWeight',
            type: 'UserWeight'
        },
        {
            id: 7,
            icon: logFast,
            title: 'Log a Fast',
            screen: 'LogFast',
            type: 'UserFast'
        },
        {
            id: 8,
            icon: logInsulin,
            title: 'Log Insulin',
            screen: 'LogInsulin',
            type: 'UserInsulin'
        }
    ],
    trackFast: {
        state: {
            tracking: 'tracking',
            tracked: 'tracked',
            complete: 'complete',
            empty: ''
        }
    },
    userDataList: [
        {
            name: 'Tamara Green',
            message: 'Hello! Did you mean this?',
            time: '14/03/2023 9:00'
        },
        {
            name: 'Adam Storm',
            message: 'Hello! Did you mean this?',
            time: '13/03/2023 08:20'
        },
        {
            name: 'Samantha Green',
            message: 'Hello! Did you mean this?',
            time: '12/03/2023 13:50'
        },
        {
            name: 'Kristen Stuart',
            message: 'Hello! Did you mean this?',
            time: '12/03/2023 10:05'
        },
        {
            name: 'Adam Storm',
            message: 'Hello! Did you mean this?',
            time: '05/03/2023 11:50'
        },
        {
            name: 'Tamara Green',
            message: 'Hello! Did you mean this?',
            time: '05/02/2023 09:50'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '06/02/2023 16:00'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '06/02/2023 15:00'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '06/02/2023 13:10'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '08/02/2023 17:50'
        },
        {
            name: 'Adriana Cole',
            message: 'Hello! Did you mean this?',
            time: '01/01/2023 10:40'
        },
        {
            name: 'Samantha Green',
            message: 'Hello! Did you mean this?',
            time: '01/01/2023 21:30'
        },
        {
            name: 'Kristen Stuart',
            message: 'Hello! Did you mean this?',
            time: '02/01/2023 08:10'
        },
        {
            name: 'Adam Storm',
            message: 'Hello! Did you mean this?',
            time: '02/01/2023 12:00'
        },
        {
            name: 'Tamara Green',
            message: 'Hello! Did you mean this?',
            time: '03/01/2023 16:00'
        },
        {
            name: 'Samantha Green',
            message: 'Hello! Did you mean this?',
            time: '06/01/2023 19:00'
        },
        {
            name: 'Kristen Stuart',
            message: 'Hello! Did you mean this?',
            time: '08/01/2023 11:40'
        }
    ],
    confirmationDialog: {
        title: {
            discard: 'Do you want to discard your changes?',
            delete: 'Are you sure you want to delete this log?'
        }
    },
    errors: {
        futureTime: 'You can not select future time.',
        fastDuration: 'The logged time should be greater than 0',
        exerciseDuration: 'The exercise duration should be greater than 0',
        logValue: 'The log value should be greater than 0'
    },
    maxFontSizeMultiplier: 1.2
};

export const customReactionData: ReactionData[] = [
    {
        Icon: Like,
        type: '+1'
    },
    {
        Icon: Heart,
        type: 'heart'
    },
    {
        Icon: Muscle,
        type: 'muscle'
    },
    // {
    //     Icon: SmileyFace,
    //     type: 'slightly_smiling_face'
    // },
    {
        Icon: SadFace,
        type: 'cry'
    },
    {
        Icon: HighFive,
        type: 'raised_hands'
    }
];
