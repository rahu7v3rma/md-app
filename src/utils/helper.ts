export const padNum = (num: number) => {
    return (num < 10 ? '0' : '') + num;
};

export const isTimerValid = (timer: number) => {
    return timer > 0;
};

export const isUnitValid = (unit: number) => {
    return unit > 0;
};

// Format number to restrict specified digits max after decimal point
export const formatNumber = (num: number, digits = 2) => {
    return +num.toFixed(digits);
};
