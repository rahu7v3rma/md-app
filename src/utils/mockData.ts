import {
    Attachment,
    Channel,
    ChannelMemberResponse,
    StreamChat,
    UserResponse
} from 'stream-chat';

import { EditLogBlood } from '@/types/log';

export const MockData: any = {
    journey: [
        {
            is_completed: false,
            order: 1,
            block: {
                name: 'cs 101',
                icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs101.png',
                locked: false,
                description:
                    'Discover how to eliminate anxiety about what you put in your mouth'
            },
            user_lessons: [
                {
                    is_completed: false,
                    order: 1,
                    lesson: {
                        id: 1,
                        title: 'programming',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/programming.png'
                    }
                },
                {
                    is_completed: true,
                    order: 2,
                    lesson: {
                        id: 2,
                        title: 'db',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/db.png'
                    }
                }
            ]
        },
        {
            is_completed: false,
            order: 2,
            block: {
                name: 'cs 102',
                icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs101.png',
                locked: false,
                description:
                    'Discover how to eliminate anxiety about what you put in your mouth'
            },
            user_lessons: [
                {
                    is_completed: false,
                    order: 1,
                    lesson: {
                        id: 1,
                        title: 'programming',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/programming.png'
                    }
                },
                {
                    is_completed: false,
                    order: 2,
                    lesson: {
                        id: 2,
                        title: 'db',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/db.png'
                    }
                }
            ]
        },
        {
            is_completed: false,
            order: 3,
            block: {
                name: 'cs 103',
                icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs102.png',
                locked: true,
                description:
                    'Discover how to eliminate anxiety about what you put in your mouth'
            },
            user_lessons: [
                {
                    is_completed: false,
                    order: 1,
                    lesson: {
                        id: 3,
                        title: 'algorithms',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/algorithms.png'
                    }
                },
                {
                    is_completed: false,
                    order: 2,
                    lesson: {
                        id: 4,
                        title: 'patterns',
                        icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/patterns.png'
                    }
                }
            ]
        }
    ],
    activeBlock: {
        is_completed: false,
        order: 1,
        block: {
            name: 'cs 101',
            icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/cs101.png',
            locked: false,
            description:
                'Discover how to eliminate anxiety about what you put in your mouth'
        },
        user_lessons: [
            {
                is_completed: false,
                order: 1,
                lesson: {
                    id: 1,
                    title: 'programming',
                    icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/programming.png'
                }
            },
            {
                is_completed: true,
                order: 2,
                lesson: {
                    id: 2,
                    title: 'db',
                    icon: 'https://bh-md-static-37458bd29d6a2cf3.s3.amazonaws.com/db.png'
                }
            }
        ]
    },
    logTabData: [
        {
            icon: '<svg> </svg> ',
            id: 1,
            screen: 'LogBlood',
            title: 'Log Blood Glucose',
            type: 'UserGlucose'
        },
        {
            id: 2,
            icon: '<svg> </svg> ',
            title: 'Log Hydration',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 3,
            icon: '<svg> </svg> ',
            title: 'Log Hydration3',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 4,
            icon: '<svg> </svg> ',
            title: 'Log Hydration4',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 5,
            icon: '<svg> </svg> ',
            title: 'Log Hydration5',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 6,
            icon: '<svg> </svg> ',
            title: 'Log Hydration6',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        },
        {
            id: 7,
            icon: '<svg> </svg> ',
            title: 'Log Hydration7',
            screen: 'LogWaterIntake',
            type: 'UserDrink'
        }
    ],
    dailyCompletedTasks: {
        UserDrink: false,
        UserExercise: false,
        UserFast: true,
        UserGlucose: true,
        UserInsulin: true,
        UserLesson: false,
        UserMedication: true,
        UserWeight: false
    },
    pickerValues: {
        hydration: {},
        weight: {},
        glucose: {},
        medication: {},
        insulin: {},
        exercise: {}
    }
};

export const mockContentReducerState = {
    loading: false,
    navigationKey: '',
    journey: [
        {
            is_completed: false,
            order: 1,
            block: {
                name: 'block1',
                icon: '/media/75beddf6-e8e1-4a0c-aaf3-7890c94f8aab.png',
                locked: false,
                description: 'description1'
            },
            user_lessons: [
                {
                    lesson: {
                        id: 1,
                        title: 'lesson1',
                        icon: '/media/b27dd75e-43fb-4200-add0-50020abe0b06.png'
                    },
                    is_completed: true,
                    order: 1
                },
                {
                    lesson: {
                        id: 2,
                        title: 'lesson2',
                        icon: '/media/332cc47d-9200-4f53-8909-b9a07f59ce00.png'
                    },
                    is_completed: false,
                    order: 2
                },
                {
                    lesson: {
                        id: 3,
                        title: 'lesson3',
                        icon: '/media/90425ab2-931b-41d3-bcd0-e44ada75c60f.png'
                    },
                    is_completed: false,
                    order: 3
                }
            ]
        }
    ]
};

export const mockContentSelector = {
    loading: false,
    activeLesson: {
        lesson: {
            id: 2,
            title: 'lesson2',
            icon: '/media/332cc47d-9200-4f53-8909-b9a07f59ce00.png'
        },
        is_completed: false,
        order: 2
    },
    activeBlock: {
        is_completed: false,
        order: 1,
        block: {
            name: 'block1',
            icon: '/media/75beddf6-e8e1-4a0c-aaf3-7890c94f8aab.png',
            locked: false,
            description: 'description1'
        },
        user_lessons: [
            {
                lesson: {
                    id: 1,
                    title: 'lesson1',
                    icon: '/media/b27dd75e-43fb-4200-add0-50020abe0b06.png'
                },
                is_completed: false,
                order: 1
            },
            {
                lesson: {
                    id: 2,
                    title: 'lesson2',
                    icon: '/media/332cc47d-9200-4f53-8909-b9a07f59ce00.png'
                },
                is_completed: true,
                order: 2
            }
        ]
    }
};

export const logItem: EditLogBlood = {
    type: 'UserLesson',
    id: 1,
    log_time: '',
    amount: 0,
    unit: '',
    measurement_type: ''
};

export const mockAttachment: Attachment = {
    image_url:
        'https://us-east.stream-io-cdn.com/1237334/images/bbc8c8f0-fd25-48b3-8967-bd256bc5eb1b.anime.png?Key-Pair-Id=APKAIHG36VEWPDULE23Q&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly91cy1lYXN0LnN0cmVhbS1pby1jZG4uY29tLzEyMzczMzQvaW1hZ2VzL2JiYzhjOGYwLWZkMjUtNDhiMy04OTY3LWJkMjU2YmM1ZWIxYi5hbmltZS5wbmc~Km9oPTIyNCpvdz0yMjQqIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjk4NzQwNTg0fX19XX0_&Signature=Rdm3s~09cXNa5NxdRZH2RGk0~MByiEytIEt6~GzkQN2cuOrNr8OPTxj1kEFFpY8rQYIIVyJ6Edp-Fz4OyWepBqEAIYXURYnof~syP4sdQT84hsowJbxuh1fQUuQPB3MVq6lxL3cKIkRFemTrSdTXS~dr6Uxmoo7xT8~5h6hp9JN2i19UWUSac-8QBeObyHvCqLPpicx3NTkQHKrjDEifrhROhWfspbaGJJPQ52D168XvGE3YgIsZhsur0sXdooVJYl4Nc2kh9tIH35cbS9d6Y5oAQsFLDFZiRgh-C1zjDroAL-F1rtZ~25mxhnlgDyN6CJNCB4R5B4-Oxehps-nGTw__&oh=224&ow=224',
    mime_type: 'image/png',
    originalFile: {
        extension: null,
        fileSize: 14380,
        filename: 'anime.png',
        height: 225,
        playableDuration: null,
        source: 'picker',
        uri: 'ph://6F90DC2A-FCC2-4512-9B37-1BC0DB175580/L0/001',
        width: 225
    },
    original_height: 224,
    original_width: 224,
    type: 'giphy'
};
export const currentUser: UserResponse = {
    id: '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528',
    role: 'customer',
    created_at: '2023-02-28T21:11:55.069783Z',
    updated_at: '2023-09-04T11:19:39.641563Z',
    last_active: '2023-10-06T17:15:26.915989907Z',
    banned: false,
    online: true,
    invisible: false,
    devices: [],
    mutes: [],
    channel_mutes: [],
    unread_count: 0,
    total_unread_count: 0,
    unread_channels: 0,
    language: '',
    name: 'Test User 1',
    image: null
};

export const members: Record<string, ChannelMemberResponse> = {
    '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528': {
        user_id:
            '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528',
        user: {
            id: '1ba586c0b89202f7307b61f1229330978a843afc98589ffc6a62f209225d3528',
            role: 'customer',
            created_at: '2023-02-28T21:11:55.069783Z',
            updated_at: '2023-09-04T11:19:39.641563Z',
            last_active: '2023-10-06T16:49:52.895391019Z',
            banned: false,
            online: true,
            name: 'Test User 1',
            image: null
        },
        created_at: '2023-03-18T18:07:35.050513Z',
        updated_at: '2023-03-18T18:07:35.050513Z',
        banned: false,
        shadow_banned: false,
        role: 'member',
        channel_role: 'channel_member'
    },
    e0bc614e4fd035a488619799853b075143deea596c477b8dc077e309c0fe42e9: {
        user_id:
            'e0bc614e4fd035a488619799853b075143deea596c477b8dc077e309c0fe42e9',
        user: {
            id: 'e0bc614e4fd035a488619799853b075143deea596c477b8dc077e309c0fe42e9',
            role: 'coach',
            created_at: '2023-02-28T20:53:38.035486Z',
            updated_at: '2023-09-04T11:19:47.822109Z',
            last_active: '2023-10-03T20:24:58.357617Z',
            banned: false,
            online: false,
            name: 'Coach Coach Test Coach 1',
            image: 'coach image'
        },
        created_at: '2023-03-18T18:07:35.050513Z',
        updated_at: '2023-03-18T18:07:35.050513Z',
        banned: false,
        shadow_banned: false,
        role: 'owner',
        channel_role: 'channel_member'
    }
};

export class ChannelGenerator {
    channel: Channel | null;
    _client: StreamChat | undefined;
    user_id: string;
    chatProfile_token: string;

    constructor(
        _client: StreamChat,
        user_id: string,
        chatProfile_token: string
    ) {
        this._client = _client;
        this.channel = null;
        this.user_id = user_id;
        this.chatProfile_token = chatProfile_token;
    }

    async initChannel() {
        if (this._client) {
            await this._client.connectUser(
                { id: this.user_id },
                this.chatProfile_token
            );
            let channels = await this._client.queryChannels({
                members: { $in: [this.user_id] }
            });
            this.channel = channels[0];
        }
    }

    setName(name: string) {
        if (this.channel?.data) {
            this.channel.data.name = name;
        }
    }

    setImage(image: string) {
        if (this.channel?.data) {
            this.channel.data.image = image;
        }
    }

    setMembers(_members: Record<string, ChannelMemberResponse>) {
        if (this.channel?.state) {
            this.channel.state.members = _members;
        }
    }
}
