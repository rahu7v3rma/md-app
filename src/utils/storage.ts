import AsyncStorage from '@react-native-async-storage/async-storage';

const FCM_PUSH_TOKEN = 'fcmPushToken';

export const getCachedFCMPushToken = async () => {
    return await AsyncStorage.getItem(FCM_PUSH_TOKEN);
};

export const setCachedFCMPushToken = async (newToken: string) => {
    await AsyncStorage.setItem(FCM_PUSH_TOKEN, newToken);
};
