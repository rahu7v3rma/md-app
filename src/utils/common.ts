import { once } from 'lodash';
import { PixelRatio, Platform } from 'react-native';

import { API_BASE_URL } from './config';

const warnLocalBackend = once(() =>
    console.warn('Working with a local backend')
);

export const COMMON = {
    get isIos() {
        return Platform.OS === 'ios';
    },
    get apiBaseUrl() {
        if (API_BASE_URL) {
            return API_BASE_URL;
        } else {
            warnLocalBackend();

            return this.isIos
                ? 'http://127.0.0.1:8000/'
                : 'http://10.0.2.2:8000/';
        }
    },
    responsiveSize(value: number) {
        return PixelRatio.getPixelSizeForLayoutSize(value) * 0.35;
    },
    stringFormat(s: string, ...args: any[]) {
        return s.replace(/{([0-9]+)}/g, (match, index) =>
            typeof args[index] === 'undefined' ? match : args[index]
        );
    },
    delay(ms: number) {
        return new Promise((resolve) => setTimeout(resolve, ms));
    },
    retryOperation(
        operation: () => Promise<any>,
        delayMs: number,
        retries: number
    ) {
        return new Promise<any>((resolve, reject) => {
            return operation()
                .then(resolve)
                .catch((reason) => {
                    if (retries > 0) {
                        return COMMON.delay(delayMs)
                            .then(
                                COMMON.retryOperation.bind(
                                    null,
                                    operation,
                                    delayMs,
                                    retries - 1
                                )
                            )
                            .then(resolve)
                            .catch(reject);
                    }

                    return reject(reason);
                });
        });
    }
};
