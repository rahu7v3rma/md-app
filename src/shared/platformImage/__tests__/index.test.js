import React from 'react';
import { Image } from 'react-native';
import { act, create } from 'react-test-renderer';

import loadingIndicatorSource from '@/assets/svg/avatar.svg';
import PlatformImage from '@/shared/platformImage';
import { COMMON } from '@/utils/common';

const imageId = 'imageId';
const width = '100px';
const height = '100px';
const method = 'scale';
const imageName = `${imageId}/r/${width}/${height}/${method}`;

jest.spyOn(console, 'warn').mockImplementation(() => {});

test('snapshot / renders without crashing', async () => {
    let tree;
    await act(() => {
        tree = create(<PlatformImage imageId={imageId} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders with specific width and height / renders with default "scale" method when no method is provided', async () => {
    let tree;
    await act(() => {
        tree = create(
            <PlatformImage imageId={imageId} width={width} height={height} />
        );
    });
    const image = tree.root.findByType(Image);
    expect(image.props.source.uri).toBe(
        `${COMMON.apiBaseUrl}image/${imageName}`
    );
});

test('renders with a custom method when provided', async () => {
    let tree;
    await act(() => {
        tree = create(
            <PlatformImage
                imageId={imageId}
                width={width}
                height={height}
                method={method}
            />
        );
    });
    const image = tree.root.findByType(Image);
    expect(image.props.source.uri).toBe(
        `${COMMON.apiBaseUrl}image/${imageName}`
    );
});

test('renders with a loading indicator source when provided', async () => {
    let tree;
    await act(() => {
        tree = create(
            <PlatformImage
                imageId={imageId}
                loadingIndicatorSource={loadingIndicatorSource}
            />
        );
    });
    const image = tree.root.findByType(Image);
    expect(image.props.loadingIndicatorSource).toBe(loadingIndicatorSource);
});
