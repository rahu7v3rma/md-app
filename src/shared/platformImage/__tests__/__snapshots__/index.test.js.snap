// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot / renders without crashing 1`] = `
<Image
  resizeMode="contain"
  source={
    {
      "headers": {
        "X-Authorization": "Token ",
      },
      "uri": "http://127.0.0.1:8000/image/imageId",
    }
  }
/>
`;
